cmds/core/sync/sync.go

```diff
@@ -1,12 +1,68 @@
-// Copyright 2016-2017 the u-root Authors. All rights reserved
+// Copyright 2016-2020 the u-root Authors. All rights reserved
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
 // sync command in Go.
+//
+// Synopsis:
+//		sync [-df] [FILE]
+//
+
 package main
 
-import "syscall"
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+)
+
+const SYS_SYNCFS = 306
+
+var (
+	data = flag.Bool("d", false, "sync file data, no metadata")
```

Comment:
and right here is why I dislike the mimic'ing GNU utilities so much. They were done this way for reasons that no longer make sense IMHO.
Were I to do this now, I'd have a verb, with valid values "data", "filesystem", shorthand "d", "f" and then do a switch below on them, and not have to do the "are they both set" test since the opcode only ever has one value. But I guess we're stuck with -d and -f ... bleah.

Reply:
Food for thought right here.

Reply:
> and right here is why I dislike the mimic'ing GNU utilities so much. They were done this way for reasons that no longer make sense IMHO.

Thank you for the valuable out-of-the-box thought pattern presented here. It made me rethinking the initial approach to the problem whilst keeping your suggestions in mind.

> Were I to do this now, I'd have a verb, with valid values "data", "filesystem", shorthand "d", "f" and then do a switch below on them, and not have to do the "are they both set" test since the opcode only ever has one value.

After refactoring the main logic now drills down to
```
        switch {
        case *data:
                doSyscall(syscall.SYS_FDATASYNC)
        case *filesystem:
                doSyscall(unix.SYS_SYNCFS)
        default:
                syscall.Sync()
                os.Exit(0)
        }
```


>But I guess we're stuck with -d and -f ... bleah.
not anymore:)
```shell
# ./sync -h
Usage: ./sync [OPTION] [FILE]...
  -d
  -data
    	sync file data, no metadata
  -f
  -filesystem
    	commit filesystem caches to disk
```